cdap-ui/app/cdap/components/DataPrepHome/index.js

```diff
@@ -112,35 +114,68 @@ export default class DataPrepHome extends Component {
     this.eventEmitter.emit('DATAPREP_CLOSE_SIDEPANEL');
   }
 
-  updateWorkspaceList() {
-    let namespace = NamespaceStore.getState().selectedNamespace;
-
+  updateWorkspaceListPoll(namespace) {
```

Comment:
This is not entirely a poll right. We are still fetching the list of workspaces here. Just that we have added the logic of retrying 3 times before failing.